### PR TITLE
Co-author UI (OCT-193)

### PR DESCRIPTION
**Pull request by nathan-at-jisc**
The purpose of this PR was to add the ability for publication owners to add co-authors. We ended up doing far more:

- When a user clicks `login`, they are now redirected to the page they were on first
- When a user lands on an authenitcated page, but are not authenticated, when they login, they are redirected to where they were
- Setting up mailhog locally
- Setting up AWS SES with DKIM
- When a user adds a co-author, and email is sent
- Should the user click "no", they will be removed from the co-author list
- Should the user click "yes", the co-author record will be linked to the logged in user
- Update user profile page to include "works"
- Update publication page to show different information if the person is a co-author
- Allow co-authors to retrieve DRAFT publications

---
### Screenshots:
Screenshot 2022-04-28 at 10 06 15
Screenshot 2022-04-28 at 10 06 22
Screenshot 2022-04-27 at 14 02 11
Screenshot 2022-04-26 at 15 38 33
Screenshot 2022-04-26 at 16 07 43

### CODE REVIEW

File: ui/src/components/Publication/Breakdown/index.tsx [hunk 1]
```diff
@@ -9,82 +10,42 @@ type Props = {
...
-                                color={value.color}
-                                width={max}
-                                className="col-span-2 mt-1 h-4 lg:col-span-1 lg:mt-0 lg:h-full"
-                            />
-                        </div>
-                    );
-                })}
-            </div>
+        <div>
+            <Chart.Radar data={data} />
```
Comment (AshleyRedmanJisc) [inline ui/src/components/Publication/Breakdown/index.tsx:48]: Do you think that we should keep the chart in for now or remove it based on the talks in the standup? If not, maybe we remove the logic & packages and take it fresh when it comes round?

File: ui/src/lib/helpers.tsx [hunk 2]
```diff
@@ -161,9 +161,13 @@ export const guardPrivateRoute = (context: Types.GetServerSidePropsContext): str
...
     if (!token) {
         context.res.writeHead(302, {
-            Location: Config.urls.orcidLogin.path
+            Location: `${Config.urls.orcidLogin.path}&state=${Buffer.from(
+                context.req.url || Config.urls.home.path,
+                'utf-8'
+            ).toString('base64')}`
         });
         context.res.end();
+        throw new Error('Sorry kid');
```
Comment (AshleyRedmanJisc) [inline ui/src/lib/helpers.tsx:170]: 😆  Possibly there is a different error message to throw?

File: api/src/components/authorization/controller.ts [hunk 3]
```diff
@@ -72,11 +72,28 @@ export const authorize = async (event: I.APIRequest<I.AuthorizeRequestBody>): Pr
             })
         );
 
+        console.log(JSON.stringify(userInformation['activities-summary'].works['group'], null, 4));
```
Comment (AshleyRedmanJisc) [inline api/src/components/authorization/controller.ts:75]: Log to remove?

File: api/src/components/coauthor/controller.ts [hunk 4]
```diff
@@ -96,6 +106,12 @@ export const link = async (
     try {
         const publication = await publicationService.get(event.pathParameters.id);
 
+        console.log({
```
Comment (AshleyRedmanJisc) [inline api/src/components/coauthor/controller.ts:109]: Log to remove?

File: ui/src/pages/authors/[id]/index.tsx [hunk 5]
```diff
@@ -66,12 +66,12 @@ const Author: Types.NextPage<Props> = (props): React.ReactElement => {
                     </h3>
                 </header>
 
-                <section id="content" className="container mx-auto px-8 pb-12 lg:pb-24">
+                {/* <section id="content" className="container mx-auto px-8 pb-12 lg:pb-24">
```
Comment (AshleyRedmanJisc) [inline ui/src/pages/authors/[id]/index.tsx:69]: Maybe we remove the commented out parts if the plan is not to include it until far time post launch?

File: ui/tailwind.config.js [hunk 6]
```diff
@@ -54,6 +54,18 @@ module.exports = {
                 800: 'hsl(331, 87%, 15%)', // #470525
                 900: 'hsl(333, 85%, 5%)' // #18020c
             },
+            green: {
```
Comment (AshleyRedmanJisc) [inline ui/tailwind.config.js:57]: hsl versions preferred to match the other colors, though it doesn't effect the site

**Review by AshleyRedmanJisc — COMMENTED** [review]
Few comments around console logs & commented out code.

The UI prettier CI has failed, so it's worth running the formatting command in the UI directory to auto format all files in `/src`. The command should be:

`npm run format:write && npm run lint` 

It's also worth running the same command in the API to ensure all code is formatted & passing the linter. Currently no CI linting for the API so having to do it manually for now. Also assuming all API tests are passing etc

File: ui/src/components/Publication/Breakdown/index.tsx (hunk 1 above)
Comment (nathan-at-jisc) [reply ui/src/components/Publication/Breakdown/index.tsx:48]: OK sweet, happy to do that yeah

File: ui/src/lib/helpers.tsx (hunk 2 above)
Comment (nathan-at-jisc) [reply ui/src/lib/helpers.tsx:170]: Yes, absolutely!

File: api/src/components/coauthor/controller.ts (hunk 4 above)
Comment (nathan-at-jisc) [reply api/src/components/coauthor/controller.ts:109]: Will address, thank you

File: ui/tailwind.config.js (hunk 6 above)
Comment (nathan-at-jisc) [reply ui/tailwind.config.js:57]: Yeah, will sort that, i was being lazy

File: api/src/components/authorization/controller.ts [hunk 7]
```diff
@@ -72,11 +72,27 @@ export const authorize = async (event: I.APIRequest<I.AuthorizeRequestBody>): Pr
             })
         );
 
+        // eslint-disable-next-line @typescript-eslint/dot-notation
```
Comment (AshleyRedmanJisc) [inline api/src/components/authorization/controller.ts:75]: This is either a interface mismatch with how it's accessing or another problem. Temp disabled this ts error.

**Review by AshleyRedmanJisc — APPROVED** [review]
Good to go

**Pull request merged**